### DISCUSSION TITLE
ChatGPT使用技巧

### DISCUSSION THREAD

**Discussion by xipeiguo** [1101-GS4719-C/Python-Community-GS4719-C#2071] [General]
https://www.cw.com.tw/article/5131709
在這次的期末專題中深有同感，常在詢問chatgpt的過程中發生我問的問題和它提供的解法無法對到的情形，這時候便要重新換種方式敘述，也是很考驗自己的溝通技巧，以及要學會如何簡化問題等，都是一大難題

**Comment by peresonacifer** [comment]
在用ChatGPT製作專題的過程中，他有時真的沒辦法根據我的問題回答我要的答案，也很常出現他給的程式一直出錯的問題，這時我真的只能從頭開始問，然後換個方向很快就成功了，從經驗上來說，問題不能太多和太複雜，最好是很好理解且有實例說明，我覺得最方便的還是他能省去自己查資料的時間，有時我還會把我對程式的理解傳給他，請他幫我修正，這樣他就能回答到我想要的

**Comment by gigitilto** [comment]
在詢問gpt問題的時候丟給他一份資料，如果問太多東西，他可能會忘記之前我之前給的那份資料以及前面回答的問題，導致回答得越來越敷衍、答非所問，而如果再丟一次資料，雖然會回答，但反應的時間感覺就變慢了許多，所以我覺得直接開始新的對話重新問會比較好。

**Comment by wcyeee** [comment]
我也覺得如何問GPT問題是一門很大的學問！（甚至有人在販賣如何問GPT問題的線上課程！）我覺得可以順便訓練自己的溝通技巧，雖然真人和ChatGPT是完全不一樣的，但因為問GPT問題需要更精準，所以我覺得透過這個練習也可以增加我們在跟真人溝通的技巧！

**Comment by Jinyuu17** [comment]
強烈建議，要完整使用 ChatGPT 的話樣購買他們的方案，雖然被 OpenAI 綁架了，但他們提供了更好的大型語言模型，可以更快更有效的提供答案。供大家參考，我已經訂閱快一年了。

**Comment by Jinyuu17** [comment]
在問問題時，可以把角色帶入，並且要有明確的指令給他，也可以給他你希望的輸出結果。以下我提供一個提問範例請教生成式 AI 一些 Python 的問題：
我是一位剛學成式語言的學生，麻煩幫我解釋 python 中 for 迴圈的使用方式，例如在...情況適合用；...情況要用別的函式。並給我一個練習範例。

**Comment by lai1008** [comment]
要完整使用 ChatGPT 的話樣購買他們的方案，雖然被 OpenAI 綁架了，但他們提供了更好更完整的大型語言模型，可以更快更有效的提供答案。

**Comment by ShuliANN** [comment]
我自己在使用chatgpt時，首先，會給予人設(叫他扮演誰+驗證是否是真的+告訴它這人設有什麼基本特徵)。接著問問題時要注意不要一次給他太多資訊，然後問問題用詞要精確，講到關鍵字，他才會給予符合要求的答覆。

**Comment by Mminniehuang** [comment]
chatgpt常常會給出它覺得我們想要的答案而不是正確答案，像是有的時候還會給出兩種回答，請我們選擇比較喜歡的答案。所以除了盡量給出精準的問題之外，在chatgpt的使用上真的要再三確認，不斷反問確認或是拆分成很多小問題輔助自己釐清整個問題。

**Comment by yue083139** [comment]
我覺得詢問chat gpt的方法也是一門很大的學問，聽說還有一門叫prompt engineering的專門學問，我覺得當中有一點提到像是給AI一個身份當成角色扮演的方法挺有趣的，還有可能就是列點敘述之類的方法，但我還是希望未來的AI可以再更加方便一點，畢竟如果我們在打prompt 上花了太多時間，好像就與我們希望快速得到知識所以使用生成式ai背道而馳了。

**Comment by christaaah** [comment]
我通常都是先概述一下情況，然後先問一個比較沒哪麼特定的問題，根據Chatgpt的回答再進行調整，看是要再問更具體一點還是補充說明內容，最後如果鬼打牆的話我就會直接開一個新對話，然後瑣碎的問題我會使用其他生成式AI或上網找資料等等，不會再問Chatgpt。

**Comment by ed112601513** [comment]
我感覺自從使用了ChatGPT厚我的溝通技巧變好了許多，我自己是很常讓ChatGPT寫程式，如果一次要它做太多東西，它可能一次會錯很多，或根本亂寫，有時候能跑出東西但根本不是我要的，所以後來我是改成分很多段問，再逐個debug和確認結果是否符合我的期望，這樣做的結果會好很多。
